tensorflow_datasets/image/caltech_birds.py

```diff
@@ -0,0 +1,183 @@
+# coding=utf-8
+# Copyright 2019 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+from collections import defaultdict
+import numpy as np
+import os
+import re
+
+_DESCRIPTION = """\
+Caltech-UCSD Birds 200 (CUB-200) is an image dataset with photos 
+of 200 bird species (mostly North American). The total number of 
+categories of birds is 200 and there are 6033 images in the dataset.
+Annotations include bounding boxes, segmentation labels.
+"""
+
+_URL = ("http://www.vision.caltech.edu/visipedia/CUB-200.html")
+_CITATION = """\
+@techreport{WelinderEtal2010,
+Author = {P. Welinder and S. Branson and T. Mita and C. Wah and F. Schroff and S. Belongie and P. Perona},
+Institution = {California Institute of Technology},
+Number = {CNS-TR-2010-001},
+Title = {{Caltech-UCSD Birds 200}},
+Year = {2010}
+}
+"""
+_IMAGES_URL = "http://www.vision.caltech.edu/visipedia-data/CUB-200/images.tgz"
+_SPLIT_URL = "http://www.vision.caltech.edu/visipedia-data/CUB-200/lists.tgz"
+_ANNOTATIONS_URL = "http://www.vision.caltech.edu/visipedia-data/CUB-200/annotations.tgz"
+_NAME_RE = re.compile(r"((\w*)/)*(\d*).(\w*)/(\w*.jpg)$")
+
+
+class CaltechBirds2010(tfds.core.GeneratorBasedBuilder):
+    '''Caltech Birds dataset'''
+
+    VERSION = tfds.core.Version("0.1.0")
+
+    def _info(self):
+
+        return tfds.core.DatasetInfo(
+                builder=self,
+                description=_DESCRIPTION,
+                features=tfds.features.FeaturesDict({
+                    # Images are of varying size
+                    # TODO: Need to add part attributes and their annotations

```

Comment:
You should check `tensorflow_datasets/image/coco.py` about todo.

Reply:
Could please elaborate on it? I looked in coco.py and did not understand it